packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Toggle from 'terra-toggle';
+import { injectIntl, intlShape } from 'terra-base';
+import IconChevronDown from 'terra-icon/lib/icon/IconChevronDown';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './ShowHide.module.scss';
+import Button from './_ShowHideButton';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Content in the body of the component that will be shown or hidden
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Text that will be visible to the user while the component is collapsed
+   */
+  preview: PropTypes.string.isRequired,
+  /**
+   * Button text that will be displayed when the component is collapsed
+   */
+  collapsedButtonText: PropTypes.string,
+  /**
+   * Button text that will be displayed when the component is expanded
+   */
+  expandedButtonText: PropTypes.string,
+  /**
+   * Icon displayed next to the button text
+   */
+  icon: PropTypes.element,
+  /**
+   * The intl object to be injected for translations. Provided by the injectIntl function.
+   */
+  intl: intlShape.isRequired,
```

Comment:
We actually are using these as part of the context instead of with the injectIntl. We don't want to expose the to the props table. https://github.com/cerner/terra-core/blob/master/packages/terra-form-field/src/Field.jsx#L81

Reply:
@StephenEsser Thanks for the example. I will give this a try

Reply:
@StephenEsser @JakeLaCombe using this.context is marked as deprecated. are we still ok with using it?
https://reactjs.org/docs/legacy-context.html

Reply:
That's what I would recommend until we have a strategy to uplift the other components, unless you're interested in pioneering the new context API, but that may have a large scope of changes to all internationalized components.

Reply:
Alright thanks for the update :)

Reply:
I think what you have (using injectIntl, getting intl through props) is what you want to do. This insulates us from the Context API entirely. If we interface with the legacy context APIs here (and I know we do elsewhere), we'll just have more changes to make in the future. react-intl should be able to passively switch to the new Context API without us noticing.